sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1774,6 +1774,50 @@ class CastSuite extends CastSuiteBase {
       assert(e3.contains("Casting 2147483648 to int causes overflow"))
     }
   }
+
+  test("SPARK-35111: Cast string to year-month interval") {
+    checkEvaluation(cast(Literal.create("INTERVAL '1-0' YEAR TO MONTH"),
+      YearMonthIntervalType), 12)
+    checkEvaluation(cast(Literal.create("0-0"), YearMonthIntervalType), 0)
+    checkEvaluation(cast(Literal.create("INTERVAL '-1-0' YEAR TO MONTH"),
+      YearMonthIntervalType), -12)
+    checkEvaluation(cast(Literal.create("INTERVAL -'-1-0' YEAR TO MONTH"),
+      YearMonthIntervalType), 12)
+    checkEvaluation(cast(Literal.create("INTERVAL +'-1-0' YEAR TO MONTH"),
+      YearMonthIntervalType), -12)
+    checkEvaluation(cast(Literal.create("INTERVAL +'+1-0' YEAR TO MONTH"),
+      YearMonthIntervalType), 12)
+    checkEvaluation(cast(Literal.create("INTERVAL +'1-0' YEAR TO MONTH"),
+      YearMonthIntervalType), 12)
+    checkEvaluation(cast(Literal.create("-1-0"), YearMonthIntervalType), -12)
+    checkEvaluation(cast(Literal.create("INTERVAL '10-1' YEAR TO MONTH"),
+      YearMonthIntervalType), 121)
+    checkEvaluation(cast(Literal.create("10-1"), YearMonthIntervalType), 121)
+    checkEvaluation(cast(Literal.create("interval '10-1' year TO MONTH"),
+      YearMonthIntervalType), 121)
+    checkEvaluation(cast(Literal.create(null, StringType), YearMonthIntervalType), null)
+
+    Seq("INTERVAL '0-0' YEAR TO MONTH", "INTERVAL '10-1' YEAR TO MONTH",
```

Comment:
We can merge the tests above into this loop
```
val hiveStyleInterval = interval.split("'")(1)
// test with both interval and hiveStyleInterval
```

Reply:
Done